src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    public Node<T> first;
+    public Node<T> last;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node previous;
+        private Node next;
+
+        public Node(Node previous, T element, Node next) {
+            this.previous = previous;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
+    private Node getNodeByIndex(int index) {
+        if (index == 0) {
+            return first;
+        }
+        if (index == size - 1) {
+            return last;
+        }
+        Node result = first;
+        for (int i = 1; i <= index; i++) {
+            result = result.next;
+        }
+        return result;
+    }
+
+    private Node getNodeByElement(T t) {
+        Node elementToSort = first;
+        while (elementToSort.next != null) {
+            if (t.equals(elementToSort.element)) {
+                return elementToSort;
+            }
+            elementToSort = elementToSort.next;
+        }
+
+        return null;
+    }
+
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size = 1;
+        } else {
+            Node node = new Node(last, value, null);
+            last.next = node;
+            last = node;
+            size++;
+        }
 
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node temp = getNodeByIndex(index);
+        if (index == 0 && size == 0) {
+            add(value);
+            size++;
```

Comment:
@TarasHryts you can move this to the end of method

Reply:
done

Reply:
@TarasHryts have you moved it?